test-dts/h.test-d.ts

```diff
@@ -149,7 +150,6 @@ describe('h support for generic component type', () => {
   function foo(bar: Component) {
     h(bar)
     h(bar, 'hello')
-    // @ts-expect-error
     h(bar, { id: 'ok' }, 'hello')
```

Comment:
After having a thought on this, I don't think having a Prop type check on `Component` is good, since if the user is using `Component` interface is because he does not know the type, therefor type checking is not needed here.

If we want type checking having a Prop is recommended 
```ts
declare function renderMyComp<P>(p: Component<P>, props: P): void
renderMyComp({} as Component<{ a: string }>, { a: '' })
```